src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -13,22 +14,39 @@ public static partial class RuntimeHelpers
         public delegate void CleanupCode(object userData, bool exceptionThrown);
 
         /// <summary>
-        /// GetSubArray helper method for the compiler to slice an array using a range.
+        /// Slices the specified array using the specified range.
         /// </summary>
         public static T[] GetSubArray<T>(T[] array, Range range)
         {
-            Type elementType = array.GetType().GetElementType();
-            Span<T> source = array.AsSpan(range);
+            if (array == null)
```

Comment:
do we need to check for null? wouldn't throw null reference exception anyway if it is null?

Reply:
From the PR description:
```
Second, it fixes argument validation so that we throw an ArgumentNullException if the input array is null rather than NullReferenceException.
```
I think it's useful to throw an ArgumentNullException that contains the details about what was null rather than a NullReferenceException that's harder to diagnose, especially if this ends up moving to live in a more prominent location.

Reply:
Yes, I saw that. I was just wondering even when getting null reference exception will have the stack that can tell where the problem is. I heard different arguments regarding the pros and cons about checking the parameters for null. I am inclining to not check it. I am ok if you think otherwise though.